mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -189,6 +190,18 @@ final class GBTClassificationModel private[ml](
 
   override def treeWeights: Array[Double] = _treeWeights
 
+  // We use two vals with options rather than lazy vals since we want
+  // the codeGen to be eagerly evaluated.
+  val treePredictors = useCodeGen match {
```

Comment:
Is there a reason to prefer match over if/else here? If/else seems more intuitive.